лаба 7

AminaMahauri opened the pull request.

AminaMahauri committed:
кр 1

билет 13

AminaMahauri committed:
Лаба 7

NemirovichMark commented:
Здравствуйте, Амина.

1й и 2й номер принял 🥇 
3й номер
Публичные сеттеры - нарушение принципов ООП
    public string Name { get; set; }
    public int GoalsScored { get; set; }
    public int GoalsConceded { get; set; }

Нужно сделать 2 массива: мужских команд и женских команд. Провести внутри них соревнования. Потом объединить в новый 3й массив все команды, отсортировать и вывести.

На доработку.

AminaMahauri committed:
Update Лаба 7

Марк Анатольевич, здравствуйте! Я исправила 3 номер: сделала два массива для мужских и женских команд, и публичные сеттеры заменила на приватные

AminaMahauri commented:
Марк Анатольевич, доброй ночи!! Вы увидели мое исправление?

NemirovichMark commented:
Здравствуйте.
Да. Это уже ближе.
Давайте поставлю более четкую задачу:
Сделать массив из 3-х отдельных команд мужчин и массив для женщин. Отсортировать команды по баллам. Создать команду победителя и ей присвоить команду с наибольшим количеством баллов.

На доработку.

AminaMahauri commented:
Уровень 3 номер 4 
using System;
using System.Collections.Generic;
using System.Linq;

public abstract class FootballTeam
{
    public  string Name { get; private set; }
    public  int GoalsScored { get; private set; }
    public int GoalsConceded { get; private set; }

    public FootballTeam(string name, int goalsScored, int goalsConceded)
    {
        Name = name;
        GoalsScored = goalsScored;
        GoalsConceded = goalsConceded;
    }

    public abstract int CalculatePoints();
}

public class MenFootballTeam : FootballTeam
{
    public MenFootballTeam(string name, int goalsScored, int goalsConceded) : base(name, goalsScored, goalsConceded) { }

    public override int CalculatePoints()
    {
        if (GoalsScored > GoalsConceded)
            return 3;
        else if (GoalsScored == GoalsConceded)
            return 1;
        else
            return 0;
    }
}

public class WomenFootballTeam : FootballTeam
{
    public WomenFootballTeam(string name, int goalsScored, int goalsConceded) : base(name, goalsScored, goalsConceded) { }

    public override int CalculatePoints()
    {
        if (GoalsScored > GoalsConceded)
            return 3;
        else if (GoalsScored == GoalsConceded)
            return 1;
        else
            return 0;
    }
}

class Program
{
    static void Main()
    {
        List<FootballTeam> menTeams = new List<FootballTeam>
        {
            new MenFootballTeam("Динамо", 4, 3),
            new MenFootballTeam("Спартак", 3, 5),
            new MenFootballTeam("Зенит", 2, 1)
        };

        List<FootballTeam> womenTeams = new List<FootballTeam>
        {
            new WomenFootballTeam("ЦСКА", 5, 2),
            new WomenFootballTeam("Локомотив", 1, 1),
            new WomenFootballTeam("Рубин", 3, 4)
        };

        List<FootballTeam> allTeams = new List<FootballTeam>();
        allTeams.AddRange(menTeams);
        allTeams.AddRange(womenTeams);

        List<MenFootballTeam> maleTeams = allTeams.OfType<MenFootballTeam>().ToList();
        List<WomenFootballTeam> femaleTeams = allTeams.OfType<WomenFootballTeam>().ToList();

        maleTeams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));
        femaleTeams.Sort((x, y) => y.CalculatePoints().CompareTo(x.CalculatePoints()));

        FootballTeam winningTeam = maleTeams[0].CalculatePoints() > femaleTeams[0].CalculatePoints() ? (FootballTeam)maleTeams[0] : (FootballTeam)femaleTeams[0];

        foreach (var team in allTeams)
        {
            System.Console.OutputEncoding = System.Text.Encoding.UTF8;
            Console.WriteLine($"{team.Name} - {team.CalculatePoints()} баллов");
        }

        Console.WriteLine($"Победитель: {winningTeam.Name}");
    }
}

AminaMahauri committed:
лаба 7 (исправление №3)

AminaMahauri commented:
Марк Анатольевич, добрый день! Я исправила 3 номер ( создала массивы и 3-х отдельных команд для мужчин и женщин, команды сортируются по баллам, потом создаёт команда победителя и ей присваивается комната с наибольшим количеством баллов, принципы ООП вроде соблюдены) Я не смогла внести исправления в существующий уже пулл, поэтому оставила коммент, но еще сделала коммит у себя в репозитории "лаба 7 (исправление №3) "

NemirovichMark commented:
Отлично! 
(Готовьтесь, что я по многим сторонним функциям на защите спрошу)
